legend-engine-language-pure-modelManager-sdlc/src/main/java/org/finos/legend/engine/language/pure/modelManager/sdlc/pure/PureServerLoader.java

```diff
@@ -44,14 +46,33 @@ public PureServerLoader(MetaDataServerConfiguration metaDataServerConfiguration)
         this.metaDataServerConfiguration = metaDataServerConfiguration;
     }
 
-    public String buildPureMetadataVersionURL(String urlSuffix)
+    public String getBaseUrl(String overrideUrl)
     {
-        return metaDataServerConfiguration.getPure().getBaseUrl() + "/alloy/pureServerBaseVersion" + urlSuffix;
+        if (overrideUrl == null || overrideUrl.equals(metaDataServerConfiguration.getPure().getBaseUrl()))
+        {
+            return metaDataServerConfiguration.getPure().getBaseUrl();
+        }
+        List<String> allowedOverrideUrls = ((PureServerConnectionConfiguration)metaDataServerConfiguration.getPure()).allowedOverrideUrls;
```

Comment:
Is this cast (PureServerConnectionConfiguration) backward compatible? I think we should check in above 'if' clause.

if (overrideUrl == null || overrideUrl.equals(metaDataServerConfiguration.getPure().getBaseUrl()) || !(metaDataServerConfiguration instanceof PureServerConnectionConfiguration))

Reply:
Added it. Providing a override url to the current version of metadata server configuration will now be a no-op.